common/param.h

```diff
@@ -90,4 +94,6 @@ typedef bool(*param_cbx)(struct command *cmd,
 				   (const jsmntok_t *)NULL,	    \
 				   (arg)) == true); })
 
-#endif /* LIGHTNING_LIGHTNINGD_PARAM_H */
+/* Special flag for 'check' which allows any parameters. */
+#define p_opt_any() "", false, NULL, NULL
```

Comment:
Nicely done. This approach crossed my mind at some point but then it was gone. Its clearly better 👍